packages/sdk/src/stores/google-datastore/store.ts

```diff
@@ -22,37 +25,61 @@ export default function <D>(kind: string, store: Datastore): SortedStore<string,
     });
   }
   async function has(id: string) {
-    try {
-      await get(id);
-      return true;
-    } catch (err) {
-      return false;
-    }
+    // horribly ineficient to actually query data to see if it exists, but cannot find a better way
+    return exists(await get(id));
   }
   async function del(id: string) {
     await store.delete(makeKey(id));
   }
   async function entries() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return [result[store.KEY].name, result];
+    }) as [string, D][];
   }
   async function values() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results;
   }
+  // all this does is map over full entries to return the id of the entry. have not found a better way to query this.
   async function keys() {
-    return [];
+    const [results] = await store.createQuery(kind).run();
+    return results.map((result) => {
+      return result[store.KEY].name;
+    }) as string[];
   }
+  // theres no way to really do built into the store client. Google recommends managing a size entry yourself.
   async function size() {
-    return 0;
+    throw new Error("size not supported in google store");
+    // typescript complains if theres no return value
+    return NaN;
```

Comment:
I think this will go away if you explicitly define the return value:
`size(): number {`